Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -2,17 +2,157 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include "Core/GeckoCodeConfig.h"
+
 #include <algorithm>
 #include <sstream>
 #include <string>
 #include <vector>
 
+#include <curl/curl.h>
+
 #include "Common/IniFile.h"
+#include "Common/Logging/Log.h"
 #include "Common/StringUtil.h"
-#include "Core/GeckoCodeConfig.h"
 
 namespace Gecko
 {
+static size_t DownloadCodesWriteCallback(void* contents, size_t size, size_t nmemb,
+                                         std::string* body)
+{
+  size_t realsize = size * nmemb;
+  body->insert(body->end(), reinterpret_cast<char*>(contents),
+               reinterpret_cast<char*>(contents) + realsize);
+  return realsize;
+}
+
+std::vector<GeckoCode> DownloadCodes(std::string gameid, bool* succeeded)
```

Comment:
I'm not sure I see the advantage of splitting out the game_id munging from the parsing. The way I read the code, this is part of the same logic of "get codes from geckocodes.org". Is this useful anywhere else?

I'm trying to think of a good way to separate out the Curl code. I guess I could do something like

```c++
  DolphinCurl curl { "http://geckocodes.org/txt.php?txt=" + gameid };
  curl->setopt(CURLOPT_TIMEOUT, 5);
  std::string response_body;
  if (!(*succeeded = curl->perform(&response_body)) {
    return {};
  };
```

What do you think?